bls/src/main/java/tech/pegasys/teku/bls/impl/blst/BlstSignature.java

```diff
@@ -41,26 +40,25 @@
   static final BlstSignature INFINITY;
 
   static {
-    p2_affine ec2Point = new p2_affine();
-    blst.p2_uncompress(ec2Point, INFINITY_BYTES.toArrayUnsafe());
+    P2_Affine ec2Point = new P2_Affine(INFINITY_BYTES.toArrayUnsafe());
     INFINITY = new BlstSignature(ec2Point, true);
   }
 
   public static BlstSignature fromBytes(Bytes compressed) {
     if (compressed.equals(INFINITY_BYTES)) {
       return INFINITY;
     }
+
     checkArgument(
         compressed.size() == COMPRESSED_SIG_SIZE,
         "Expected " + COMPRESSED_SIG_SIZE + " bytes of input but got %s",
         compressed.size());
-    p2_affine ec2Point = new p2_affine();
+    P2_Affine ec2Point = null;
     try {
-      BLST_ERROR rc = blst.p2_uncompress(ec2Point, compressed.toArrayUnsafe());
-      return new BlstSignature(ec2Point, rc == BLST_ERROR.BLST_SUCCESS);
+      ec2Point = new P2_Affine(compressed.toArrayUnsafe());
+      return new BlstSignature(ec2Point, true);
     } catch (Exception e) {
-      ec2Point.delete();
-      throw e;
+      return new BlstSignature(new P2_Affine(), false);
```

Comment:
The reason I did this was that we used to use the `blst.p2_uncompress` method in the past, and if it returned an error, we'd just create the Signature as invalid. Now there is no static method, and the P2_Affine constructor throws an error instead of returning a `BLST_ERROR` enum. Thus, I had to return an invalid signature instance in the exception case, to not break tests.

Reply:
Oh, then totally makes sense 👍